### ISSUE TITLE
Counting the number of elements written by the encoder

### ISSUE THREAD

**Issue by jeanphilippelingrand** [mtth/avsc#347]
Hi there! This is not really an "issue" but more a call for help/solution for a given problem I'm trying to solve.

# I'd like to count the number of items (objects) the file encoder has written. I'm using `avro.createFileEncoder(pathToFile, schema,opt)` to create the avro file.

# What I've tried
- I've plugged a listener to the 'data' event thinking each element written to the file would trigger the event but looks like it's not a 1-1 relation.
- I also have been trying to tweak the block size parameter of the createFileEncoder opt. Since the doc says:

>   blockSize {Number} Maximum uncompressed size of each block data.
>   A new block will be started when this number is exceeded.
>   If it is too small to fit a single element, it will be increased appropriately. Defaults to 64kB.

I've tried setting the value to 1 thinking that each element written in the file would trigger the 'data' event.

# Voila
Let me know if you have any idea how to get this count info. If this is not the right place for this kind of post lmk i'll close this issue. 
Thanks!

**Comment by mtth** [comment]
Hi @jeanphilippelingrand. The number of written values was not accessible from the encoder itself. You would have had to intercept the writes before, for example within an earlier piped stream. However since this seems generally useful, I released a new version of `avsc` which gives you access to the information you need via a new `'block'` event.

Sample usage:

```js
let count = 0;
const encoder = createFileEncoder(fp, schema)
  .on('block', (info) => {
    count += info.valueCount;
  })
  .on('finish', () => {
    console.log(count); // Total number of values encoded.
  });
// Write values...
```

`info` above contains three fields:

+ `valueCount`, the number of values in that block (this is what you want);
+ `rawDataSize`, the size of the block before compression;
+ `compressedDataSize`, the size of the block after compression, if any.

This event is available starting with version `5.6.3`.

Finally, to explain why what you tried didn't work:

+ Block encoder (what `createFileEncoder` uses under the hood) `'data'` events emit "container blocks" which can include multiple values. For performance and efficiency reasons the number of blocks is typically much lower than the number of values.
+ Since the input `blockSize` is increased when it is too small to fit a single encoded value, multiple smaller values encountered later might fit within an increased size block.

**Comment by jeanphilippelingrand** [comment]
Thank you @mtth for this crystal clear explanation.
